tests/test_gcloud_store.py

```diff
@@ -64,6 +67,39 @@ def try_delete_bucket(bucket):
         assert err.message.endswith("unknown error")
 
 
+def get_bucket_from_store(gcstore: GoogleCloudStore) -> Bucket:
+    cached_bucket = getattr(gcstore, f"{LAZY_PROPERTY_ATTR_PREFIX}bucket", None)
+    if cached_bucket:
+        return cached_bucket
+
+    client = get_client_from_store(gcstore)
+    if gcstore.create_if_missing and not client.lookup_bucket(gcstore.bucket_name):
+        bucket = client.create_bucket(
+            bucket_or_name=gcstore.bucket_name,
+            location=gcstore.bucket_creation_location,
+        )
+    else:
+        # will raise an error if bucket not found
+        bucket = client.get_bucket(gcstore.bucket_name)
+
+    setattr(gcstore, f"{LAZY_PROPERTY_ATTR_PREFIX}bucket", bucket)
+    return bucket
+
+
+def get_client_from_store(gcstore: GoogleCloudStore) -> Client:
+    cached_client = getattr(gcstore, f"{LAZY_PROPERTY_ATTR_PREFIX}client", None)
+    if cached_client:
+        return cached_client
+
+    if type(gcstore._credentials) == str:
+        client = Client.from_service_account_json(gcstore._credentials)
+    else:
+        client = Client(credentials=gcstore._credentials, project=gcstore.project_name)
+
+    setattr(gcstore, f"{LAZY_PROPERTY_ATTR_PREFIX}client", client)
```

Comment:
why are you setting the client (and the bucket, for that matter) as attributes inside the store?

Reply:
I did that to avoid recreating the `Bucket` object every time. It's probably a relatively cheap operation, so maybe we can also skip that caching.